data_utils.py

```diff
@@ -152,21 +194,86 @@ def prepare_transform(self, label_vals):
 
     def convert_to_raw_label_scale(self, dst_l_cut, preds):
         if (self.i2cat is None) or (self.scalers_dict is None):
-            raise RuntimeError("self.i2cat or self.scalers_dict is None. Run ScaleUtil.transform_df function first")
+            raise RuntimeError(
+                "self.i2cat or self.scalers_dict is None. Run ScaleUtil.transform_df function first")
         raw_preds = []
         for dst, pred in zip(dst_l_cut, preds):
             cat = self.i2cat[dst]
             if self.scale_label == 'Cbrt':
                 raw = np.power(pred, 3)
                 raw_preds.append(raw)
             else:
-                raw = self.scalers_dict[cat].inverse_transform(pred.reshape(1, -1))
+                raw_pred = self.scalers_dict[cat].inverse_transform([pred.tolist()])  # Convert numpy array to a list
                 if self.scale_label == 'Log':
-                    raw = np.sign(raw) * (np.exp(np.abs(raw))-1)
+                    raw_pred = np.sign(raw_pred) * (np.exp(np.abs(raw_pred)) - 1)
+                    
                 elif self.scale_label == 'Cbrt':
-                    raw = np.power(raw, 3)
-                raw_preds.append(raw[0, 0])
+                    raw_pred = np.power(raw_pred, 3)
+                    raw_preds.append(raw_pred[0][0])  # Access the value from the list
         if isinstance(preds, np.ndarray):
             return np.array(raw_preds)
         else:
-            return torch.tensor(raw_preds, dtype=float, device=self.device)
\ No newline at end of file
+            return torch.tensor(raw_preds, dtype=float, device=self.device)
+
+
+    def get_centroid_preds(self,dst_l_cut,preds):
```

Comment:
delete this and let `convert_to_raw_label_scale(self, dst_l_cut, preds)` to handle it.  `self.scalers_dict[cat].inverse_transform([pred.tolist()])` in `convert_to_raw_label_scale` will do the trick. I'll fix the `inverse_transform` method in `KBinsDiscretizer`